app/models/banner.rb

```diff
@@ -0,0 +1,27 @@
+class Banner < ActiveRecord::Base
+  enum record_type: { "category": 0, "item": 1 }
+
+  after_create :generate_url
+
+  belongs_to :bannerable, polymorphic: true
+
+  scope :recent, -> { order(id: :DESC) }
+
+  mount_uploader :image, OriginalPicUploader
+
+  private
+
+  def generate_url
+    baes_url = "https://www.mmwooo.com"
```

Comment:
url do not create like this
you should call url_helper , like item_url(category, item)